pkg/controller/common/keystore/user_secret.go

```diff
@@ -45,19 +46,30 @@ func secureSettingsVolume(
 ) (*volume.SecretVolume, string, error) {
 	// setup (or remove) watches for the user-provided secret to reconcile on any change
 	watcher := k8s.ExtractNamespacedName(hasKeystore)
-	if err := watches.WatchUserProvidedSecrets(
+
+	// user-provided Secrets referenced in the resource
+	secretSources := WatchedSecretNames(hasKeystore)
+	// user-provided Secrets referenced in a StackConfigPolicy that configures the resource
+	policySecretSources, err := filesettings.GetSecureSettingsSecretSources(ctx, r.K8sClient(), hasKeystore)
```

Comment:
Will this not break keystone implementations in Kibana and Beats and APM Server that don't have the file-based secure settings secret ?